Core/OxidEE_Events.php

```diff
@@ -200,46 +197,66 @@ private static function _deleteOrderTransactionTable()
     {
         $sQuery = "DROP TABLE IF EXISTS `wdoxidee_ordertransactions`";
 
-        $oDb = oxDb::getDb();
-        $oDb->Execute($sQuery);
+        self::$oDb->Execute($sQuery);
+    }
+
+    /**
+     * Add Wirecard's payment methods defined in payments.xml
+     */
+    private static function _addPaymentMethods()
+    {
+        $oLogger = Registry::getLogger();
+        $oConfig = Registry::getConfig();
+        $sShopBaseURL = $oConfig->getShopUrl();
+        $xmldata = simplexml_load_file($sShopBaseURL . "modules/wirecard/paymentgateway/payments.xml");
+        if ($xmldata) {
+            foreach ($xmldata->payment as $payment) {
```

Comment:
Please, outsource this code block into a separate private function called `_addPaymentMethod`, thus increasing overall readability.